### PR TITLE
re org registry

### PR DESCRIPTION
**Pull request by wangxinyi7**
### Description

In this PR, we lift the `registry` out of the server code so that client, server and cli can share the registry. It's stored in `BaseDeps.Deps` (not `setup.go` but `options.go`) because when we newServer, we are passing `baseDeps.Deps` instead of `baseDeps`.

This [PR](https://github.com/hashicorp/consul/pull/18052/files) shows how we use the registry when register the http endpoints.

Notice I changed the signature of functions: `setupExternalGRPC` and `setupInternalResourceService`. Please let me know if there are better options.

### PR Checklist

* [ ] ~updated test coverage~
* [ ] ~external facing docs updated~
* [ ] ~appropriate backport labels added~
* [ ] ~not a security concern~

### CODE REVIEW

**Review by mkeeler — APPROVED** [review]

**Review by JadhavPoonam — APPROVED** [review]
Let's clean up the commit history before we merge 🚢

File: agent/consul/client_test.go [hunk 1]
```diff
@@ -7,6 +7,7 @@ import (
 	"bytes"
 	"context"
 	"fmt"
+	"github.com/hashicorp/consul/internal/resource"
```
Comment (loshz) [inline agent/consul/client_test.go:10]: NIT: we need to run `goimports` on this file.

File: agent/consul/server.go [hunk 2]
```diff
@@ -8,6 +8,7 @@ import (
 	"crypto/x509"
 	"errors"
 	"fmt"
+	"github.com/hashicorp/consul/internal/resource"
```
Comment (loshz) [inline agent/consul/server.go:11]: `goimports`

File: agent/rpc/peering/service_test.go [hunk 3]
```diff
@@ -8,6 +8,7 @@ import (
 	"encoding/base64"
 	"encoding/json"
 	"fmt"
+	"github.com/hashicorp/consul/internal/resource"
```
Comment (loshz) [inline agent/rpc/peering/service_test.go:11]: `goimports`

**Review by loshz — COMMENTED** [review]
Overall looks good. Just need to run `goimports` on a couple of the files 👍

File: agent/consul/client_test.go (hunk 1 above)
Comment (wangxinyi7) [reply agent/consul/client_test.go]: Thanks for spotting this out! Configured my editor so that it will run `goimports` files automatically everytime

**Comment by wangxinyi7** [comment]
> Let's clean up the commit history before we merge 🚢

Actually the merging strategy right now is set as `squash` so current commit messages won't be recorded. Only the PR title and description are picked by default and we could change them when we press the button.

**Pull request merged**